tests/test_sftp_automatic.py

```diff
@@ -0,0 +1,123 @@
+from base import TestSFTPBase
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+import os
+import csv
+import json
+
+class TestSFTPAutomatic(TestSFTPBase):
+
+    def name(self):
+        return "tap_tester_sftp_automatic_fields"
+
+    def get_files(self):
+        return [
+            {
+                "headers": ['id', 'string_col', 'integer_col'],
+                "directory": "table_1_files",
+                "files": ["table_1_fileA.csv"],
+                "num_rows": 50,
+                "generator": self.generate_simple_csv_lines_typeA
+            }
+        ]
+
+    def setUp(self):
+        if not all([x for x in [os.getenv('TAP_SFTP_USERNAME'),
```

Comment:
Setup is also used in many files so move this to base.py and for diff lines, use super and add those lines.

Reply:
The SetUp() function is related to pytest so it will require in each file to set up something before running that particular test.